packages/connect-wallet-modal/src/connectButtons/connectMetamask.tsx

```diff
@@ -7,8 +7,21 @@ import ConnectButton from './connectButton';
 import checkConflicts from './checkConflicts';
 
 const ConnectMetamask: FC<ConnectWalletProps> = (props: ConnectWalletProps) => {
-  const { onConnect, shouldInvertWalletIcon, setRequirements, ...rest } = props;
-  const { connect } = useConnectorMetamask();
+  const {
+    onConnect,
+    shouldInvertWalletIcon,
+    setRequirements,
+    metrics,
+    ...rest
+  } = props;
+  const onConnectMetamask =
+    metrics?.events?.connect?.handlers.onConnectMetamask;
+  const { connect } = useConnectorMetamask({
+    onConnect: () => {
+      onConnect?.();
+      onConnectMetamask?.();
```

Comment:
There are a lot of code duplicates, is there any way to pass onConnect* from outside somehow?

Reply:
Well, we have a lot of duplicates across all the current reef-knot codebase for now, and this is bad. But honestly, I don't want to do a further refactoring right now.

Reply:
Sure thing, just wanted to note it